pkg/providers/gcp/provider_postgres.go

```diff
@@ -19,14 +25,30 @@ import (
 	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
 	v1 "k8s.io/api/core/v1"
 	k8serr "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	utils "k8s.io/utils/pointer"
 )
 
 const (
-	postgresProviderName         = "gcp-cloudsql"
-	ResourceIdentifierAnnotation = "resourceIdentifier"
-	defaultCredSecSuffix         = "-gcp-sql-credentials"
+	postgresProviderName                          = "gcp-cloudsql"
+	ResourceIdentifierAnnotation                  = "resourceIdentifier"
+	defaultCredSecSuffix                          = "-gcp-sql-credentials"
+	defaultGCPCLoudSQLDatabaseVersion             = "POSTGRES_13"
+	defaultGCPCloudSQLRegion                      = "us-central1"
+	defaultGCPIdentifierLength                    = 40
+	defaultGCPPostgresUser                        = "postgres"
+	defaultPostgresPasswordKey                    = "password"
+	defaultPostgresUserKey                        = "user"
+	defaultTier                                   = "db-custom-2-3840"
+	defaultAvailabilityType                       = "REGIONAL"
+	defaultStorageAutoResizeLimit                 = 100
+	defaultStorageAutoResize                      = true
+	defaultBackupConfigEnabled                    = true
+	defaultPointInTimeRecoveryEnabled             = true
+	defaultBackupRetentionSettingsRetentionUnit   = "COUNT"
+	defaultBackupRetentionSettingsRetainedBackups = 30
+	defaultDataDiskSizeGb                         = 20
```

Comment:
Might be worth checking if we can get away with smaller instance storage size. The minimum in the UI is 10GB 🤔

Reply:
These defaults were decided on by @adam-cattermole

Reply:
Discussed and decision made to leave default at 20GB.